setup.py

```diff
@@ -60,6 +66,57 @@ class Components:
 except Exception as e:
     raise RuntimeError("Could not get version from Git repo : " + str(e))
 
+# Check if we've got --cmake-build-dir path that will be used to build extensions only
+# and not to build NEURON wheels.
+just_extensions = False
+cmake_build_dir = None
+if "--cmake-build-dir" in sys.argv:
+    cmake_build_dir = sys.argv[sys.argv.index("--cmake-build-dir") + 1]
+    sys.argv.remove("--cmake-build-dir")
+    sys.argv.remove(cmake_build_dir)
+    # Check that the CMake build directory is valid
+    if not os.path.exists(cmake_build_dir):
+        raise RuntimeError(
+            "CMake build directory does not exist: {}".format(cmake_build_dir)
+        )
+    # Check that the CMake build directory has been configured and built
+    if not os.path.exists(os.path.join(cmake_build_dir, "CMakeCache.txt")):
+        raise RuntimeError(
+            "CMake build directory does not contain CMakeCache.txt: {}".format(
+                cmake_build_dir
+            )
+        )
+    # check for libnrniv.{so,dylib, dll.a} depending on platform
+    check_suffix = "so" if sys.platform != "darwin" else "dylib"
+    if sys.platform == "win32":
+        check_suffix = "dll.a"
+    if not os.path.exists(
+        os.path.join(
+            cmake_build_dir,
+            "lib",
+            "libnrniv.{}".format(check_suffix),
+        )
+    ):
+        raise RuntimeError(
+            "CMake build directory does not contain libnrniv: {}".format(
+                cmake_build_dir
+            )
+        )
+    just_extensions = True
+
+# Check for RX3D Optimization Level
+rx3d_opt_level = "-O0"
+if "--rx3d-opt-level" in sys.argv:
```

Comment:
This seems very pedestrian to me, why not something along the lines of https://stackoverflow.com/questions/18725137/how-to-obtain-arguments-passed-to-setup-py-from-pip-with-install-option ?

Reply:
setup.py install is deprecated, don't want to go that route

Reply:
OK. Seems that install options are also present in other build systems for Pÿthon, though

Reply:
Yeah, but I always fail to see consistency/correlation